src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs

```diff
@@ -353,11 +352,29 @@ private static void AddRoslynOutputFilePaths(ProjectInfo project, ProjectData pr
 
         private void FixSarifAndEncoding(IList<ProjectInfo> projects, AnalysisProperties analysisProperties)
         {
-            var globalSourceEncoding = GetSourceEncoding(analysisProperties, new SonarScanner.MSBuild.Common.EncodingProvider());
+            var globalSourceEncoding = GetSourceEncoding(analysisProperties);
+            var logIfGlobalEncodingIsIgnored = () => logger.LogInfo(Resources.WARN_PropertyIgnored, SonarProperties.SourceEncoding);
```

Comment:
Could you add please a test to `PropertiesFileGeneratorTests.cs` that verifies that this action is indeed invoked and the correct logs are printed?

Reply:
All tests verify the action is or not called.

Reply:
I addee UTs. What you say is integration testing, and the reason why this code was not tested at all was difficulty to mock everything at IT level

Reply:
And indeed, this is testing at Unit level, not integration. However, adding integration tests for this case is quite difficult and outside the scope of this PR

Reply:
I can open a PR to add  ITs

Reply:
Ok, I am fine with adding an issue for it. Thanks a lot! I will approve this PR.